nilearn/mass_univariate/_utils.py

```diff
@@ -140,6 +130,15 @@ def calculate_tfce(
     return tfce_4d
 
 
+def _return_score_threshs(arr3d, dh, two_sided_test):
+    """Compute list of score threshold to use for TFCE."""
+    max_score = (
+        np.nanmax(np.abs(arr3d)) if two_sided_test else np.nanmax(arr3d)
+    )
+    number_steps = 100 if dh == "auto" else max(1, round(max_score / dh))
```

Comment:
should probably set a minimum number of steps, no?

Reply:
I don't know.  If the user has bothered to set `dh` to other than `"auto"`, maybe they actually want that value... as part of plot across multiple values, for testing purposes, ...?

Reply:
I think that in general it is better to make each step more robust without making too many assumoptions about the previous ones.

Maybe nilearn will not give you data with nans in them but some other software may (I know that SPM gives outputs 'beta images' padded with nans for examples), so I'd say it makes sense to do this.

Reply:
@bthirion 
any opinion on this?

Reply:
I think it's a good idea to use nanmax

Reply:
and any suggestion if we should a minimum number of steps for TFCE procedure?

Reply:
I would impose at least 10.

Reply:
@Leengit do you want to add a check that we got at least 10 steps and throw a warning that the number of steps will be set to 10 when it's not the case?

Reply:
throwing a warning could be done a bit like this: https://github.com/nilearn/nilearn/pull/5179/files#r1983395313

but indeed we would need to add a test that ensure that the warning is triggered: 

example of checking if a warning is thrown when running a test:
https://github.com/nilearn/nilearn/blob/8517db819c43c75f7401e18c60b5108749d12b33/nilearn/tests/test_masking.py#L267-L268